src/NServiceBus.Core/Hosting/HostInfoSettings.cs

```diff
@@ -40,7 +39,7 @@ public HostInfoSettings UsingInstalledFilePath()
         public HostInfoSettings UsingCustomIdentifier(Guid id)
         {
             Guard.AgainstNull(nameof(id), id);
-            config.Settings.Set(HostInformationFeature.HostIdSettingsKey, id);
+            config.Settings.Get<HostingComponent.Configuration>().HostId = id;
```

Comment:
The hosting component is not yet created (since it we can only create it once configuration is done) so we can access it yet. A static property would not work either since statics prevents hosting multiple endpoints in the same process/appdomain.

Does that make sense?